orcas/l1l2.go

```diff
@@ -81,8 +80,30 @@ func (l *L1L2Orca) Set(req common.SetRequest) error {
 
 	if err != nil {
 		metrics.IncCounter(MetricCmdSetErrorsL1)
-		metrics.IncCounter(MetricCmdSetErrors)
-		return err
+
+		metrics.IncCounter(MetricCmdSetL1ErrorDeleteL1)
+
+		// in order to ensure consistency, attempt a delete from L1
+		// For keys that are unable to be set in L1 but were successfully set in
+		// L2 this may cause a shift in load. These keys tend to be large so this
+		// will probably put a significant burden on L2 if the data are large.
+		// Note that even if there's a major problem, e.g. the connection being
+		// closed, this will still return success.
+		dcmd := common.DeleteRequest{
+			Key: req.Key,
+		}
+
+		start = timer.Now()
+		err = l.l1.Delete(dcmd)
+		metrics.ObserveHist(HistDeleteL1, timer.Since(start))
+
+		if err == common.ErrKeyNotFound {
+			metrics.IncCounter(MetricCmdSetL1ErrorDeleteMissesL1)
+		} else if err != nil {
+			metrics.IncCounter(MetricCmdSetL1ErrorDeleteErrorsL1)
```

Comment:
So right now, we're returning success, but really don't know what's going to happen on the next request.

I'm not sure what the type of errors we can expect on Delete operation.  If it's only memcached crash/disconnect, I think returning error here should be safe.